osi_hostvehicledata.proto

```diff
@@ -271,4 +284,184 @@ message HostVehicleData
+            ASSIST_FEATURE_OTHER = 1;
+
+            // Blind spot warning.
+            //
+            ASSIST_FEATURE_BLIND_SPOT_WARNING = 2;
+
+            // Forward collision warning.
+            //
+            ASSIST_FEATURE_FORWARD_COLLISION_WARNING = 3;
+
+            // Lane departure warning.
+            //
+            ASSIST_FEATURE_LANE_DEPARTURE_WARNING = 4;
+
+            // Parking collision warning.
+            //
+            ASSIST_FEATURE_PARKING_COLLISION_WARNING = 5;
+
+            // Rear cross-traffic warning
+            //
+            ASSIST_FEATURE_REAR_CROSS_TRAFFIC_WARNING = 6;
+
+            // Automatic emergency braking
+            //
+            ASSIST_FEATURE_AUTOMATIC_EMERGENCY_BRAKING = 7;
+
+            // Emergency steering
+            //
+            ASSIST_FEATURE_AUTOMATIC_EMERGENCY_STEERING = 8;
+
+            // Reverse automatic emergency braking
+            //
+            ASSIST_FEATURE_REVERSE_AUTOMATIC_EMERGENCY_BRAKING = 9;
+
+            // Adaptive cruise control
+            //
+            ASSIST_FEATURE_ADAPTIVE_CRUISE_CONTROL = 10;
+
+            // Lane keeping assist
+            //
+            ASSIST_FEATURE_LANE_KEEPING_ASSIST = 11;
+
+            // Active driving assistance
+            //
+            ASSIST_FEATURE_ACTIVE_DRIVING_ASSISTANCE = 12;
+
+            // Backup camera
+            //
+            ASSIST_FEATURE_BACKUP_CAMERA = 13;
+
+            // Surround view camera
+            //
+            ASSIST_FEATURE_SURROUND_VIEW_CAMERA = 14;
+
+            // Active parking assistance
+            //
+            ASSIST_FEATURE_ACTIVE_PARKING_ASSISTANCE = 15;
+
+            // Remote parking assistance
+            //
+            ASSIST_FEATURE_REMOTE_PARKING_ASSISTANCE = 16;
+
+            // Trailer assistance
+            //
+            ASSIST_FEATURE_TRAILER_ASSISTANCE = 17;
+
+            // Automatic high beams
+            //
+            ASSIST_FEATURE_AUTOMATIC_HIGH_BEAMS = 18;
+
+            // Driver monitoring
+            //
+            ASSIST_FEATURE_DRIVER_MONITORING = 19;
+
+            // Head up display
+            //
+            ASSIST_FEATURE_HEAD_UP_DISPLAY = 20;
+
+            // Night vision
+            //
+            ASSIST_FEATURE_NIGHT_VISION = 21;
+        }
+
+        // The activation state of a feature.
+        //
+        // \note Not all of these will be applicable for all vehicles
+        // and features.
+        //
+        enum ActivationState
+        {
+            // An unknown activation state, this should not be used.
+            //
+            ACTIVATION_STATE_UNKNOWN = 0;
+
+            // Used for custom states not covered by the definitions below.
+            //
+            // A string state can be specified in custom_activation_state.
+            //
+            ACTIVATION_STATE_OTHER = 1;
+
+            // The feature has been disabled.
+            //
+            ACTIVATION_STATE_TURNED_OFF = 2;
```

Comment:
I did consider this, but I preferred "turned off" as being more obvious that it has been disabled, as opposed to "off" because it hasn't been triggered yet.

I don't feel strongly though if consensus goes the other way.

Reply:
Sorry for the late replay.
I understand your use case. What I see here is, that it is a passive function, so it is always on standby.
Looking to another example makes clear what is missing:
Think about an lane keeping function doing lateral control. There the function has to be activated first.
If you are driving beside the road the function should have a state like `ACTIVATION_STATE_UNAVAILABLE`. Back on the road again it should be `_ACTIVATABLE`/`_AVAILABLE`.
Then the driver pushes a button and function goes to `_ACTIVE`. 
If the driver overrides and forces a lane change, I can agree on a state like `_DRIVER_OVERRIDE`. But it could also be `_STANDBY`. It stays there until the Lane Keeping Function finds back between the laneBoundaries and goes `_ACTIVE` again.
If now the driver pushes again the button, the function goes back to `_ACTIVATABLE`/`_AVAILABLE`. But it is not turned off completely! I think we should do a meeting to dive deeper next year, because we have to bring these two perspectives together. I'm sure we can do that :)

Reply:
CCB Meeting 02.02.2022:
- included with last commit